src/ast/operator.rs

```diff
@@ -86,41 +88,59 @@ pub enum BinaryOperator {
     PGRegexIMatch,
     PGRegexNotMatch,
     PGRegexNotIMatch,
+    PGCustomBinaryOperator(PGCustomOperator),
+}
+
+/// PostgreSQL-specific custom operator.
+///
+/// See [CREATE OPERATOR](https://www.postgresql.org/docs/current/sql-createoperator.html) for more information.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct PGCustomOperator {
+    pub schema: Option<String>,
+    pub name: String,
```

Comment:
Thank you for trying -- Something about hard coding "schema" seems not right to me here -- at the very least because there can also be a database in identifiers `https://www.postgresql.org/docs/current/ddl-schemas.html`  -- e.g. `database.schema.table`, so maybe it should be a `Vec<String>`

I think given the special set of allowed characters described in https://www.postgresql.org/docs/current/sql-createoperator.html in `The operator name is a sequence of up to NAMEDATALEN-1 (63 by default) characters from the following list:` to properly parse this limitation some extra code would be needed

Let me give it a shot